RestCommClient/Classes/Push/PushHandler.m

```diff
@@ -0,0 +1,269 @@
+                                    if (credentialsSid){
+                                        //check existing binding sid
+                                        [self checkBindingSidWithCompletionHandler:^(NSString *bindingSid) {
+                                            //create the binding object
+                                            Binding *binding = [[Binding alloc] initWithIdentity:clientSid
+                                                                         applicationSid:applicationSid
+                                                                             andAddress:token];
+                                            
+                                            //its and existing sid, we should update sid
+                                            if (bindingSid && bindingSid.length > 0){
+                                                [pushCommManager updateBinding:binding forBindingSid:bindingSid andCompletionHandler:^(NSString *bindingSid, NSError *error) {
+                                                    if (error){
+                                                        RCLogError([error.localizedDescription UTF8String]);
+                                                    }
+                                                    //save binding sid
+                                                    [[NSUserDefaults standardUserDefaults] setObject:bindingSid forKey:kBindingSidKey];
+                                                }];
+                                            } else {
+                                                [pushCommManager createBinding:binding andCompletionHandler:^(NSString *bindingSid, NSError *error) {
+                                                    if (error){
+                                                        RCLogError([error.localizedDescription UTF8String]);
+                                                    } else {
+                                                        //save binding sid
+                                                        [[NSUserDefaults standardUserDefaults] setObject:bindingSid forKey:kBindingSidKey];
+                                                    }
+                                                }];
+                                            }
+                                        }];  
+                                    }
+                                }];
+                            }
+                        }];
+                    }
+                }];
+            }
+        }];
+        
+    });
+
+}
+
+- (void)getAccountSidWithCompletionHandler:(void (^)(NSString *accountSid))completionHandler{
+    //check is account id is already saved in user defaults
+    NSUserDefaults* appDefaults = [NSUserDefaults standardUserDefaults];
+    NSString *accountSid = [appDefaults objectForKey:kAccountSidKey];
+    if (accountSid && accountSid.length > 0){
```

Comment:
Is there a chance that the accountSid exists in the 'cache' but is not valid and we need to handle that case?

Reply:
Good point. Solution will be to have combination saved:
accountSid and rescommAccountEmail; in this case, we will be sure that we have saved the account sid for email. Also, we must ask server side is the account sid persistant. wdyt?

Reply:
I don't have full picture on this implementation-wise, so use your best judgement. Just wanted to point out to make sure that cache stays up to date during configuration changes